core/src/main/scala/kafka/server/BrokerToControllerChannelManager.scala

```diff
@@ -362,24 +351,6 @@ class BrokerToControllerRequestThread(
     }
   }
 
-  override def doWork(): Unit = {
-    if (activeControllerAddress().isDefined) {
-      super.pollOnce(Long.MaxValue)
-    } else {
-      debug("Controller isn't cached, looking for local metadata changes")
-      controllerNodeProvider.get() match {
-        case Some(controllerNode) =>
-          info(s"Recorded new controller, from now on will use broker $controllerNode")
-          updateControllerAddress(controllerNode)
-          metadataUpdater.setNodes(Seq(controllerNode).asJava)
```

Comment:
I feel metadataUpdater.setNodes would be a necessary step for networkClient if the controller address is changed.

Reply:
At runtime, this metadataUpdater will be an ManualMetadataUpdator.
In this case, since each request given to the NetworkClient already contains the destination node (resolved from the controllerNodeProvider), there is no need to query the MetadataUpdator. Thus, updating the MetadataUpdater with the node list is an unnecessary operation.